cmd/kots/cli/velero.go

```diff
@@ -830,93 +836,100 @@ func VeleroConfigureHostPathCmd() *cobra.Command {
+		opts.IsMinioDisabled = isMinioDisabled
+	}
+
 	deployOptions := snapshot.FileSystemDeployOptions{
 		Namespace:        opts.Namespace,
 		IsOpenShift:      k8sutil.IsOpenShift(clientset),
 		FileSystemConfig: opts.FileSystemConfig,
 		ForceReset:       opts.ForceReset,
 	}
-	if err := snapshot.DeployFileSystemMinio(ctx, clientset, deployOptions, *opts.RegistryOptions); err != nil {
-		if _, ok := errors.Cause(err).(*snapshot.ResetFileSystemError); ok {
-			log.FinishSpinnerWithWarning(color.New(color.FgHiRed))
-			forceReset := promptForFileSystemReset(log, err.Error())
-			if forceReset {
-				log.ActionWithSpinner("Re-configuring File System Minio")
-				deployOptions.ForceReset = true
-				if err := snapshot.DeployFileSystemMinio(ctx, clientset, deployOptions, *opts.RegistryOptions); err != nil {
-					log.FinishSpinnerWithError()
-					return errors.Wrap(err, "failed to force deploy file system minio")
-				}
-			}
-		} else {
-			log.FinishSpinnerWithError()
-			return errors.Wrap(err, "failed to deploy file system minio")
-		}
-	}
-
-	log.FinishSpinner()
-	log.ActionWithSpinner("Waiting for File System Minio to be ready")
-
-	err = k8sutil.WaitForDeploymentReady(ctx, clientset, opts.Namespace, snapshot.FileSystemMinioDeploymentName, time.Minute*2)
-	if err != nil {
-		log.FinishSpinnerWithError()
-		return errors.Wrap(err, "failed to wait for file system minio")
-	}
 
-	log.FinishSpinner()
-	log.ActionWithSpinner("Creating Default Bucket")
+	if !opts.IsMinioDisabled {
+		// Minio Case
+		err = deployVeleroMinioFileSystem(ctx, clientset, log, deployOptions, opts)
+		if err != nil {
+			return errors.Wrap(err, "could not deploy minio fs")
+		}
+	} else {
+		// LVP Case
+		// Peak to see if this is a legacy minio deployment that was migrated
+		isLegacyMinioDeployment, _, err := snapshot.ValidateFileSystemDeployment(ctx, clientset, deployOptions, *opts.RegistryOptions)
+		if err != nil {
+			return errors.Wrap(err, "could not validate lvp file system")
+		}
+		opts.IsLegacyDeployment = isLegacyMinioDeployment
 
-	err = snapshot.CreateFileSystemMinioBucket(ctx, clientset, opts.Namespace, *opts.RegistryOptions)
-	if err != nil {
-		log.FinishSpinnerWithError()
-		return errors.Wrap(err, "failed to create default bucket")
+		if err := snapshot.DeployFileSystemLvp(ctx, clientset, deployOptions, *opts.RegistryOptions); err != nil {
+			return errors.Wrap(err, "could not deploy lvp file system config")
+		}
 	}
 
-	log.FinishSpinner()
-
-	veleroNamespace, err := snapshot.DetectVeleroNamespace(ctx, clientset, opts.Namespace)
+	veleroStatus, err := snapshot.DetectVelero(ctx, opts.Namespace)
 	if err != nil {
 		return errors.Wrap(err, "failed to detect velero namespace")
 	}
-	if veleroNamespace == "" {
-		c, err := buildPrintableFileSystemVeleroConfig(ctx, clientset, opts.Namespace)
+
+	if (veleroStatus == nil && !opts.IsMinioDisabled) || (veleroStatus != nil && !veleroStatus.ContainsPlugin("plugin-for-aws") && !opts.IsMinioDisabled) {
```

Comment:
it's a bit hard to figure out what this condition is trying to check for at first. i get it, but it took me a bit